Test/patterns/PatternMatching.dfy

```diff
@@ -0,0 +1,89 @@
+// RUN: %dafny /compile:3 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*
+*  This file contains a collection of tests for features modified or introduced in PR 458
+*/
+
+
+datatype Alt = A(int) | B(int)
+datatype MyOption = Some(v: Alt) | None
+datatype MyPair = Pair(x:Alt, y:Alt)
+datatype List<T> = Nil | Cons(head: T, tail: List)
+
+// Nested Patterns
+method NestingTest (xs:List<int>) returns (r:int)
+{
+    match xs
+     case Cons(y, Cons(z, zs)) => return z;
+     case Cons(y, Nil) => return y;
+     case Nil => return 0;
+}
+
+
+// Ordered match with variables
+method NestedVariableTest(x:List<int>) returns (r:int)
+{
+    match x {
+        case Cons(a, Cons(b, tl1)) => r := 0;
+        case Cons(c, tl2) => r:=1;
+        case d => r := 2;
+    }
```

Comment:
Use 2 space for each indent. (Note, since this `match` uses braces around the `case`s, each `case` is indented 2 more spaces than the `match` -- as it already is, except for using 2 spaces.)